src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1956,6 +1956,17 @@ public unsafe string TrimStart(params char[]? trimChars)
             }
         }
 
+        public string RemoveStart(string value) => RemoveStart(value, StringComparison.Ordinal);
+
+        public unsafe string RemoveStart(string value, StringComparison comparisonType)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+            if (!this.StartsWith(value, comparisonType) || value.Length > Length)
+                return this;
+            return CreateTrimmedString(value.Length, Length - 1);
```

Comment:
Does this need to use actual length matches for non-ordinal comparison, like what is suggested in https://github.com/dotnet/runtime/issues/14386#issuecomment-673777660 ?

Reply:
Yes, for anything other than ordinal it must use the match length.

Reply:
Addendum: the approved API also has changes to `CompareInfo`. The intent is that the bulk of the logic is implemented within the new `CompareInfo` APIs, and the `string` and `MemoryExtensions` additions are helper wrappers around the `CompareInfo` implementation.